pkg/filesystem/filesystem.go

```diff
@@ -241,14 +244,131 @@ func UnarchiveTgz(fs billy.Filesystem, tgzPath, tgzSubdirectory, destPath string
 		if h.Name == "pax_global_header" {
 			continue
 		}
-		return fmt.Errorf("Encountered unknown type of file (name=%s) when unarchiving %s", h.Name, tgzPath)
+		return fmt.Errorf("encountered unknown type of file (name=%s) when unarchiving %s", h.Name, tgzPath)
 	}
 	if len(tgzSubdirectory) > 0 && !subdirectoryFound {
-		return fmt.Errorf("Subdirectory %s was not found within the folder outputted by the tgz file", tgzSubdirectory)
+		return fmt.Errorf("subdirectory %s was not found within the folder outputted by the tgz file", tgzSubdirectory)
 	}
 	return nil
 }
 
+// CompareTgz
+func CompareTgzs(fs billy.Filesystem, leftTgzPath string, rightTgzPath string) (bool, error) {
+	// Read left
+	leftFile, err := fs.OpenFile(leftTgzPath, os.O_RDONLY, os.ModePerm)
+	if err != nil {
+		return false, err
+	}
+	defer leftFile.Close()
+	leftGzipReader, err := gzip.NewReader(leftFile)
+	if err != nil {
+		return false, fmt.Errorf("unable to read gzip formatted file: %s", err)
+	}
+	defer leftGzipReader.Close()
+	leftTarReader := tar.NewReader(leftGzipReader)
+
+	// Read right
+	rightFile, err := fs.OpenFile(rightTgzPath, os.O_RDONLY, os.ModePerm)
+	if err != nil {
+		return false, err
+	}
+	defer rightFile.Close()
+	rightGzipReader, err := gzip.NewReader(rightFile)
+	if err != nil {
+		return false, fmt.Errorf("unable to read gzip formatted file: %s", err)
+	}
+	defer rightGzipReader.Close()
+	rightTarReader := tar.NewReader(rightGzipReader)
```

Comment:
[tar.NewReader](https://pkg.go.dev/archive/tar?utm_source=gopls#NewReader) does not support `Close` since it takes in the `io.Reader` that will run the close.